provider/provider_program_test.go

```diff
@@ -96,6 +96,10 @@ func testProgram(t *testing.T, dir string) {
 	if testing.Short() {
 		t.Skipf("Skipping in testing.Short() mode, assuming this is a CI run without credentials")
 	}
+	// Skipping due to known flakes - see https://github.com/pulumi/pulumi-pagerduty/issues/541
+	if !(dir == "test-programs/index_usercontactmethod") {
```

Comment:
I'd prefer a skip list so that newly added programs are not skipped.
```suggestion
	if slices.Contains([]string{
		"test-programs/index_service",
		"test-programs/index_schedule",
		"test-programs/index_usernotificationrule",
		"test-programs/index_escalationpolicy",
	}, dir) {
```